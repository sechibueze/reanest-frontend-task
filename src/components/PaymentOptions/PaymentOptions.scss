@import "../../assets//theme/colors.scss";
.payment-wrapper {
  display: none;
  padding: 1rem;
  margin: 1rem auto;
  border-radius: 0.8rem;
  grid-template-columns: 1fr auto;

  background-color: $white;
  @media screen and (min-width: $mobile-breakpoint) {
    & {
      display: grid;
    }
  }
}
.p-intro {
  align-self: center;
  font-size: 0.85rem;
  font-weight: bold;
}
.payment-options-container {
  justify-content: center;
  display: flex;
  flex-direction: row;
  align-self: center;
}
.arrow {
  align-self: center;
  padding: 0rem 1rem;
  color: $blue;
}
.payment-icons {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  align-self: stretch;
  color: $greyed-out;
  opacity: 0.8;
}

.payment-icons li {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 0rem 1rem;
}

.po-name {
  font-size: 0.5rem;
  text-align: center;
  color: #444;
  margin-top: 0.5rem;
}
