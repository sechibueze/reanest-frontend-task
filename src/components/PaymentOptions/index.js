import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { Fragment } from "react";
import "./PaymentOptions.scss";
const PaymentOptions = ({ billPayment }) => {
  const icons = [
    ["far", "handshake"],
    ["far", "bookmark"],
    ["far", "credit-card"],
    ["fas", "certificate"],
    ["far", "gem"],
    ["fas", "angle-right"],
  ];
  return (
    <div className="payment-wrapper">
      <span className="p-intro">
        <Fragment> {billPayment.title} </Fragment>
        <FontAwesomeIcon
          icon={["fas", "info-circle"]}
          size="0.25x"
          style={{
            color: "white",
            backgroundColor: "blue",
          }}
        />
      </span>
      <div className="payment-options-container">
        <span className="arrow">
          <FontAwesomeIcon icon={["fas", "angle-left"]} size="2x" />
        </span>
        <ul className="payment-icons">
          {billPayment.items.map((item, idx) => {
            const { id, title } = item;

            return (
              <li key={id}>
                <span>
                  {" "}
                  <FontAwesomeIcon icon={icons[idx]} size="2x" />{" "}
                </span>
                <span className="po-name"> {title && title} </span>
              </li>
            );
          })}
        </ul>
      </div>
    </div>
  );
};

export default PaymentOptions;
