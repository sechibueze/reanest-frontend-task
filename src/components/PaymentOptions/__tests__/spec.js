import { shallow } from "enzyme";
import PaymentOptions from "../index";
const paymentChannelsCount = 6;
describe("<PaymentOptions /> renders correctly", () => {
  const component = shallow(<PaymentOptions />);
  const paymentOpts = component.find(".payment-icons li");
  const introText = component.find(".p-intro");
  it("<PaymentOptions /> matches snapshot", () => {
    expect(component).toMatchSnapshot();
  });
  it("<PaymentOptions /> shows all payment channels", () => {
    expect(paymentOpts.length).toEqual(paymentChannelsCount);
  });
  it("Should contain MAKE PAYMENT element", () => {
    expect(introText).toBeTruthy();
  });
});
