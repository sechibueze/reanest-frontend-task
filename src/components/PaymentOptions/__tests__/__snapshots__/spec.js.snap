// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentOptions /> renders correctly <PaymentOptions /> matches snapshot 1`] = `ShallowWrapper {}`;
